src/slices/comments.slice.ts

```diff
@@ -0,0 +1,61 @@
+/* eslint-disable no-param-reassign */
+import { createAsyncThunk, createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { Comment } from '../types/Comment';
+import { getPostComments } from '../api/comments';
+
+type Comments = {
+  loaded: boolean;
+  hasError: boolean;
+  items: Comment[];
+};
+
+const initialState: Comments = {
+  loaded: false,
+  hasError: false,
+  items: [],
+};
+
+export const fetchPostComments = createAsyncThunk<Comment[], number>(
+  'comments/fetchPostComments',
+  async (postId: number) => {
+    const result = await getPostComments(postId);
+
+    return result;
+  },
+);
+
+const commentsSlice = createSlice({
+  name: 'usersList',
+  initialState,
+  reducers: {
+    addNewComment: (state, action: PayloadAction<Comment>) => {
+      state.items = [...state.items, action.payload];
+    },
+    deleteComment: (state, action: PayloadAction<number>) => {
+      state.items = state.items.filter(el => el.id !== action.payload);
+    },
+    takeError: state => {
+      state.hasError = true;
+    },
+  },
+  extraReducers: builder => {
+    builder.addCase(fetchPostComments.pending, state => {
+      state.loaded = true;
```

Comment:
Setting 'state.loaded' to true in the 'pending' case might be misleading. Typically, 'loaded' should indicate that data has been successfully loaded. Consider using a different state property, such as 'loading', to indicate that a request is in progress.